### PR TITLE
 feat(files): use mocha 6's way of resolving files

### LINKED ISSUE #1428: Support `--files` in mocha.opts
I'm trying to run Stryker on a Vue project with webpack and after fixing some other issues -mainly npm package errors- I'm stuck with this error:

`Error: [vue-test-utils]: window is undefined, vue-test-utils needs to be run in a browser environment. 
You can run the tests in node using jsdom`

I've created a quick-and-dirty example at [https://github.com/glezalex/vue-tdd](https://github.com/glezalex/vue-tdd) to show an example of this error, in case somebody could help me understand where this issue comes from :).

### PR DESCRIPTION
**Pull request by nicojs**
Use mocha 6's implementation to resolving files.
This adds support for `exclude`, `spec`, `files` as well as positional arguments.

Fixes #1428 

Note: #1511 should be merged first

### CODE REVIEW

**Comment by nicojs** [comment]
This is now ready for review. 

@simondel I'm using https://github.com/mochajs/mocha/blob/master/lib/cli/run-helpers.js#L132 This is a yet undocumented API. Do you think we should work around this?

File: packages/mocha-runner/src/MochaOptionsLoader.ts [hunk 1]
```diff
@@ -5,41 +5,61 @@ import { tokens, commonTokens } from '@stryker-mutator/api/plugin';
...
-      const args = serializeArguments(overrides);
-      const rawConfig = LibWrapper.loadOptions(args) || {};
-      if (this.log.isTraceEnabled()) {
-        this.log.trace(`Mocha: ${LibWrapper.loadOptions.name}([${args.map(arg => `'${arg}'`).join(',')}]) => ${JSON.stringify(rawConfig)}`);
-      }
-      return filterConfig(rawConfig);
+      this.log.debug('Mocha >= 6 detected. Using mocha\'s `%s` to load mocha options', LibWrapper.loadOptions.name);
+      return this.loadMocha6Options(overrides);
     } else {
       this.log.debug('Mocha < 6 detected. Using custom logic to parse mocha options');
```
Comment (simondel) [inline packages/mocha-runner/src/MochaOptionsLoader.ts]: Didn't we want to log this as a deprecation warning? Or is that somewhere else?

File: packages/mocha-runner/src/MochaOptionsLoader.ts [hunk 2]
```diff
@@ -5,41 +5,61 @@ import { tokens, commonTokens } from '@stryker-mutator/api/plugin';
...
+
+  private loadMocha6Options(overrides: MochaOptions) {
+    const args = serializeArguments(overrides);
+    const loadOptions = LibWrapper.loadOptions || (() => ({}));
+    const rawConfig = loadOptions(args) || {};
+    if (this.log.isTraceEnabled()) {
+      this.log.trace(`Mocha: ${loadOptions.name}([${args.map(arg => `'${arg}'`).join(',')}]) => ${JSON.stringify(rawConfig)}`);
+    }
+    const options = filterConfig(rawConfig);
+    if (this.log.isDebugEnabled()) {
```
Comment (simondel) [inline packages/mocha-runner/src/MochaOptionsLoader.ts]: Why do you check if debug is enabled? in the `loadMochaOptions` above, you don't check this.

File: packages/mocha-runner/src/MochaOptionsLoader.ts (hunk 1 above)
Comment (nicojs) [reply packages/mocha-runner/src/MochaOptionsLoader.ts]: Let's do that in a separate PR, before we release 2.0

File: packages/mocha-runner/src/MochaOptionsLoader.ts (hunk 2 above)
Comment (nicojs) [reply packages/mocha-runner/src/MochaOptionsLoader.ts]: Because we do `JSON.stringify`. An action that's only needed if debug is enabled.

Comment (simondel) [reply packages/mocha-runner/src/MochaOptionsLoader.ts]: Check, so it's just a performance improvement.

File: packages/mocha-runner/src/MochaOptionsLoader.ts (hunk 1 above)
Comment (simondel) [reply packages/mocha-runner/src/MochaOptionsLoader.ts]: Sure 👍

**Pull request merged**